internal/notifier/datadog.go

```diff
@@ -0,0 +1,185 @@
+	}, nil
+}
+
+func (d *DataDog) Post(ctx context.Context, event eventv1.Event) error {
+	dataDogEvent := d.toDataDogEvent(&event)
+
+	_, _, err := d.eventsApi.CreateEvent(d.dataDogCtx(ctx), dataDogEvent)
+	if err != nil {
+		return fmt.Errorf("failed to post event to DataDog: %w", err)
+	}
+
+	return nil
+}
+
+// dataDogCtx returns a context with the DataDog API key set.
+// This is one way to authenticate with the DataDog API.
+func (d *DataDog) dataDogCtx(ctx context.Context) context.Context {
+	return context.WithValue(ctx, datadog.ContextAPIKeys, map[string]datadog.APIKey{
+		"apiKeyAuth": {
+			Key: d.apiKey,
+		},
+	})
+}
+
+// toDataDogEvent converts an eventv1.Event to a datadogV1.EventCreateRequest.
+func (d *DataDog) toDataDogEvent(event *eventv1.Event) datadogV1.EventCreateRequest {
+	return datadogV1.EventCreateRequest{
+		// Note: Title's printf format matches other events from datadog's kubernetes integration
+		Title: fmt.Sprintf("Events from the %s %s/%s", event.InvolvedObject.Kind, event.InvolvedObject.Name, event.InvolvedObject.Namespace),
+		Text:  event.Message,
+		Tags:  d.toDataDogTags(event),
+		// fluxcd matches the name datadog picked for their flux integration: https://docs.datadoghq.com/integrations/fluxcd/
+		SourceTypeName: strPtr("fluxcd"),
+		DateHappened:   int64Ptr(event.Timestamp.Unix()),
+		AlertType:      toDataDogAlertType(event),
+	}
+}
+
+// toDataDogTags parses an eventv1.Event to return a slice of tags.
+// We set env and kube_cluster_name to the value that the user sets for `channel`.
+// We set kind, name, and namespace to the appropriate values of the involved object.
+func (d *DataDog) toDataDogTags(event *eventv1.Event) []string {
+	// Note: Datadog's built in kubernetes tagging is documented here: https://docs.datadoghq.com/containers/kubernetes/tag/?tab=containerizedagent#out-of-the-box-tags
+	tags := []string{
+		fmt.Sprintf("flux_reporting_controller:%s", event.ReportingController),
+		fmt.Sprintf("flux_reason:%s", event.Reason),
+		// "env" is a standard tag for datadog and usually matches the cluster name: https://docs.datadoghq.com/getting_started/tagging/unified_service_tagging/?tab=kubernetes
+		fmt.Sprintf("env:%s", d.env),
+		// Note: DataDog standardizes kubernetes tags as "kube_*": https://github.com/DataDog/datadog-agent/blob/82dc933aa86de037c70fe960384aa06a62e457a8/pkg/collector/corechecks/cluster/kubernetesapiserver/events_common.go#L48
+		fmt.Sprintf("kube_kind:%s", event.InvolvedObject.Kind),
+		fmt.Sprintf("kube_name:%s", event.InvolvedObject.Name),
+		fmt.Sprintf("kube_namespace:%s", event.InvolvedObject.Namespace),
+	}
+
+	// set the cluster name to the value of the summary field if it exists
+	// otherwise, set it to the value of the channel
+	if v, ok := event.Metadata["summary"]; ok {
+		tags = append(tags, fmt.Sprintf("kube_cluster_name:%s", v))
+		delete(event.Metadata, "summary")
+	} else {
+		tags = append(tags, fmt.Sprintf("kube_cluster_name:%s", d.env))
+	}
```

Comment:
I would not do this, if people want `kube_cluster_name` they should set in the Alert e.g.

```yaml
apiVersion: notification.toolkit.fluxcd.io/v1beta2
kind: Alert
metadata:
  name: <name>
spec:
  eventMetadata:
    kube_cluster_name: "my-cluster"
    kube_cluster_region: "my-region"
```

Let's use channel for env, all the rest should come from the Alert.